### DISCUSSION TITLE
Controlling a custom Zigbee end device

### DISCUSSION THREAD

**Discussion by Faustino-ai** [Koenkk/zigbee2mqtt#6759] [General]
Anyone here knows how to send raw hex code to an Zigbee end device? 

The reason I want to do this is because I can control my custom zigbee end device with my custom zigbee coordinator by sending raw hex code (transparent transmission) but I want to know how can I do it with my CC2530 through zigbee2mqtt. Thank you

**Comment by VertorixAU** [comment]
This is a project I have been working on over the last year.  It allows a Atmega328P or Atmega32u4 to communicate with a CC2530.  The devices will be on sale soon.  You can use the library to send messages to the CC2530 over SPI or UART.

Regards
Michael

https://github.com/VertorixAU/Vertorix_VT1100_Mini
https://github.com/VertorixAU/Vertorix_VT1300_Shield
https://github.com/VertorixAU/ArduinoSketchUploader
https://github.com/VertorixAU/VertorixBoards

**Reply by Faustino-ai** [reply]
Oh wow that's nice. I will try to figure out how it works. Thanks!

**Reply by VertorixAU** [reply]
Use the VT1300 library if you want to send data to CC2530 using serial (UART).  Use the examples in the library to get started.

**Reply by Faustino-ai** [reply]
Oh I actually want to send via CC2530. It is my coordinator now

**Reply by VertorixAU** [reply]
You need to send a AF_DATA_REQUEST to the CC2530 ZNP.  See the Z-Stack ZNP Interface Specification for a description of how to format the Hex bytes.  You could use node-red to send hex bytes as a suggestion.

**Reply by Faustino-ai** [reply]
Oh ok thats good to know. I will try out node red then

**Reply by natemason** [reply]
@VertorixAU are you still developing this project? You're repo links appear to be dead now.
